source/common/http/conn_pool_base.cc

```diff
@@ -97,11 +97,15 @@ void HttpConnPoolImplBase::onPoolReady(Envoy::ConnectionPool::ActiveClient& clie
 // side we do 2^29.
 static const uint64_t DEFAULT_MAX_STREAMS = (1 << 29);
 
-void MultiplexedActiveClientBase::onGoAway(Http::GoAwayErrorCode) {
+void MultiplexedActiveClientBase::onGoAway(Http::GoAwayErrorCode error_code) {
   ENVOY_CONN_LOG(debug, "remote goaway", *codec_client_);
   parent_.host()->cluster().trafficStats()->upstream_cx_close_notify_.inc();
   if (state() != ActiveClient::State::Draining) {
     if (codec_client_->numActiveRequests() == 0) {
+      if (codec_client_->protocol() == Protocol::Http3 && error_code == GoAwayErrorCode::Other) {
+        // This must be network change because QUIC GOAWAY frame doesn't have error code.
+        close_after_network_change_ = true;
```

Comment:
Hm. I see. And is `onNetworkChanged()` the only caller which passes in `Other`? This feels a big like spooky action at a distance. We could potentially add an explicit argument to `onGoaway()` or add a new enum value, though both are possibly used in lots of places. I don't want to block if we're sure this works, but it feels a bit sketchy.

Reply:
Hm... What if we add a new interface for ActiveClient to implement. This new interface could have an `onNetworkChanged()` method.  Then `QuicHttpClientConnectionImpl` can take an instance of this interface (which would be ActiveClient, of course) where it can call some new `setNetworkChangedCallback()` method on the `EnvoyQuicClientSession`. Would that work?